save-common/src/commonMain/kotlin/org/cqfn/save/core/plugin/PluginConfig.kt

```diff
@@ -9,10 +9,31 @@ import kotlinx.serialization.Serializable
 /**
  * Core interface for plugin configuration
  */
-interface PluginConfig
+interface PluginConfig<T : PluginConfig<T>> {
+    /**
+     * Create new config by merging of [this] and corresponding config from [childConfig]
+     *
+     * @param childConfig list of child configs, which will be filtered for merging
+     * @return new config
+     */
+    @Suppress("TYPE_ALIAS")
+    fun createNewPluginConfig(childConfig: MutableList<PluginConfig<*>>): T
```

Comment:
```suggestion
    fun createNewPluginConfig(childConfig: List<PluginConfig<*>>): T
```

Reply:
For now, we modify childConfig in this method, so I left mutable list